Add Features Support to sm.ini

Dracrius opened the pull request:
### Description
Patched in the enhanced feature support used in previous snesrev projects to allow configuration of Super Metroid's extra features through the ini. Also allowed for the use of both a sm.smc or sm.sfc rom.

### Will this Pull Request break anything? 
No

### Suggested Testing Steps
Double check replay compatibility and that all features enable correctly. I personally don't have a save that allows me to test all of the features I just tested with some debug outputs.

Dracrius committed:
Made the rom check allow for both sfc and smc file extensions

Setup support for [Features] section of ini with current features
- AmmoRechargeStation : Missile stations will refill super missiles and power bomb
- ShinesparkControl : Control the shinespark in any direction
- ChainSpark : Resume speed boosting from slope
- LowHealthBeep : Lower volume of low health beep
- PowerBombReveal : Power bombs reveal super missile blocks
- InstantPickups : Quick pickups

Based on code implementation LowHealthBeep could be changed from a bool to a value and used to set how much to lower the volume by.

Dracrius committed:
Setup Replay memory patch for enhanced features.

Dracrius committed:
Changed features.h name to extra_features.h to avoid the same possible issue noted in merge #3

LaserEyess commented on src/CMakeLists.txt:
```diff
@@ -52,7 +52,7 @@ target_sources(sm PRIVATE
     "snes/snes_other.c"
     "snes/spc.c"
-)
+ "extra_features.h")
```
Headers don't need to be included in `target_sources()`, it's added below by `target_include_directories()`

Dracrius replied:
Silly VS added that and I just didn't pay attention it is now removed.

LaserEyess commented on src/extra_features.h:
```diff
@@ -0,0 +1,37 @@
+    //kFeatures0_ = 8192,
+    //kFeatures0_ = 16384,
+    //kFeatures0_ = 32768,
+    //kFeatures0_ = 65536,
```
Not that it really matters, but I feel like `1 << $i` would be a bit more readable for this

Dracrius replied:
We discussed further with @testyourmine on discord and decided to go with hex values for now. These have been changed to hex.

Dracrius committed:
Removed unneeded include in target_sources() added by VS

Dracrius committed:
Updated kRam_Features0 to an unused location (Previous was never read but was written to)
Swapped Feature enum assignment to hex values

Dracrius committed:
Changed LowHealthBeep to be a Percentage of the set value in the ini and fixed the features address to one that has room for a uint32

Dracrius committed:
Fixed assumption of 240 extended y usage and restored the ExtendedY ini setting for when someone does get around to fixing extended y

Dracrius committed:
Separated Shinespark Health feature from Shinespark_Control and added a new setting to the ini for it.

Dracrius committed:
The config.c update required for the last push

Dracrius committed:
Changed ShinesparkHealth check to a return if true

Dracrius committed:
Moved `uint32 g_wanted_sm_features;` from sm_rtl.h to sm_rtl.c. The original placement caused build issues when creating a switch build.

Dracrius committed:
Implemented the use of the Audio settings used in the ini. Also raised the default Audio Frequency to match the old hardcoded one.

Dracrius committed:
Fixed mistakes in sm_84.c (intantpickup variable instead of powerbombreveal)
Hardcoded Switch builds to use fullscreen to avoid stretched widescreen from windowed mode
